app/code/Bulbulatory/Recomendations/Model/Grid.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Bulbulatory\Recomendations\Model;
+
+use Magento\Framework\Model\AbstractModel;
+use Magento\Framework\DataObject\IdentityInterface;
+
+class Grid extends AbstractModel implements IdentityInterface
+{
+	const CACHE_TAG = 'bulbulatory_recomendations';
+	protected $_cacheTag = 'bulbulatory_recomendations';
+	protected $_eventPrefix = 'bulbulatory_recomendations';
```

Comment:
Nazwa klasy z modelem powinna odpowiadać obiektowi, który ona opisuje. Mamy w module tabelę `bulbulatory_recomendations`, która przechowuje pojedyncze rekomendacje i ich właśnie ten model dotyczy. W związku z tym klasa powinna się nazywać np. `Recommendation`, podobnie jak klasa od Resource Model oraz folder, w którym znajduje się `Collection.php`.

Nie jest to błąd, bo przecież wszystko działa. Ale wyobraźmy sobie, że w module dodamy jeszcze jedną tabelę, np. `bulbulatory_discounts` która przechowuje wykorzystania kodów zniżkowych za polecenia. Będziemy chcieli też wyświetlić to jako grid w panelu. Wtedy klasy modelu nie nazwiemy już `Grid`, bo będzie ta nazwa zajęta i trzeba będzie kombinować. Jeśli z kolei nazwiemy klasy odpowiednio `Recommendation` oraz `Discount`, wszystko będzie czytelniejsze i od razu wiadomo, który model za co odpowiada. Podobnie gdy będziemy chcieli zapisywać polecenia w bazie - będziemy się odwoływać do modelu, więc dobrze gdyby ten model rzeczywiście nazywał się tak, jak struktura, którą tworzymy.

Tak samo `CACHE_TAG`, `$_cacheTag` i `$_eventPrefix` powinny się odnosić do opisywanej struktury. Dobrze, aby składały się one z nazwy modułu oraz nazwy modelu. Nazwa modułu już jest, więc dopisanie `_recommendation` będzie wystarczające. Tutaj problemem to nie będzie, ale gdyby pojawiły się kolejne modele w tym module, to wtedy w łatwy sposób będzie można je rozróżnić.